app/api/call_for_papers.py

```diff
@@ -0,0 +1,106 @@
+from flask_rest_jsonapi import ResourceDetail, ResourceList, ResourceRelationship
+from marshmallow_jsonapi.flask import Schema, Relationship
+from marshmallow_jsonapi import fields
+from app.models import db
+from app.api.helpers.permissions import jwt_required
+from sqlalchemy.orm.exc import NoResultFound
+from flask_rest_jsonapi.exceptions import ObjectNotFound
+from app.models.event import Event
+from app.models.call_for_paper import CallForPaper
+import pytz
+
+
+class CallForPaperSchema(Schema):
+    """
+    Api Schema for cfp model
+    """
+    class Meta:
+        """
+        Meta class for cfp Api Schema
+        """
+        type_ = 'call_for_paper'
+        self_view = 'v1.call_for_paper_detail'
+        self_view_kwargs = {'id': '<id>'}
+
+    id = fields.Str(dump_only=True)
+    announcement = fields.Str()
+    start_date = fields.DateTime(nullable=True)
+    end_date = fields.DateTime(nullable=True)
+    timezone = fields.Str(default="UTC")
+    hash = fields.Str()
+    privacy = fields.String()
+    event = Relationship(attribute='event',
+                         self_view='v1.call_for_paper_event',
+                         self_view_kwargs={'id': '<id>'},
+                         related_view='v1.event_detail',
+                         related_view_kwargs={'call_for_paper_id': '<id>'},
+                         schema='EventSchema',
+                         type_='event')
+
+
+class CallForPaperList(ResourceList):
+    """
+    List and create cfp
+    """
+
+    def query(self, view_kwargs):
+        query_ = self.session.query(CallForPaper)
+        if view_kwargs.get('event_id') is not None:
+            query_ = query_.join(Event).filter(Event.id == view_kwargs['event_id'])
+        return query_
+
+    def before_create_object(self, data, view_kwargs):
+        if view_kwargs.get('event_id') is not None:
+            event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
+            start_date = data['start_date'].replace(tzinfo=pytz.timezone(data['timezone']))
+            end_date = data['end_date'].replace(tzinfo=pytz.timezone(data['timezone']))
+            print start_date.isoformat(), end_date.isoformat()
```

Comment:
@niranjan94 @shubham-padia @SaptakS I've tried doing this to make timezone aware for the date fields. While I print start_date and end_date, I am getting perfect result like this:
![zzzz](https://cloud.githubusercontent.com/assets/20799954/26751888/80d436be-4861-11e7-9a9f-5bbf197df0c5.jpg)
But when i do a GET, I get different dates.:
![screenshot from 2017-06-03 13-09-51](https://cloud.githubusercontent.com/assets/20799954/26751906/dfae9cce-4861-11e7-9e14-81a494ebd322.png)

Is there a solution for this ?